aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPlugin.kt

```diff
@@ -397,9 +394,15 @@ internal class RealAWSCognitoAuthPlugin(
                     onSuccess,
                     onError
                 )
-                is AuthenticationState.SignedIn -> onSuccess.accept(
-                    AuthSignInResult(true, AuthNextSignInStep(AuthSignInStep.DONE, mapOf(), null))
-                )
+                is AuthenticationState.SignedIn -> {
+                    onError.accept(
+                        AuthException(
```

Comment:
Should we use `SignedInException` instead?

Reply:
@div5yesh I believe the messaging here aligns with iOS. SignedInException messaging is close, but in this instance, we would lose messaging parity.

Reply:
We can change the messaging in SignedInException if it doesn't affect parity for any other cases where the exception is used.